src/library/scala/annotation/meta/constructor.scala

```diff
@@ -0,0 +1,19 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2002-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+package scala.annotation.meta
+
+/**
+ * An annotation marked with `@constructor` signals that an annotation can be
+ * placed at a constructor.
+ *
+ * Additionally, marking an annotation with `@constructor` moves an annotation
+ * usage from a class declaration to the class' primary constructor if the
+ * allowed targets of the annotation in question doesn't also include class
+ * declarations.
```

Comment:
I think we don't have a notion of "allowed targets", right? Also, is the semantics of the `@constructor` meta-annotation actually implemented in this PR?